openassessment/workflow/models.py

```diff
@@ -352,24 +356,74 @@ def _get_steps(self):
             steps = list(self.steps.all())
         return steps
 
-    def set_score(self, score):
+    def set_staff_score(self, score, staff_name, is_required=False, is_override=False, reason=None):
         """
-        Set a score for the workflow.
+        Set a staff score for the workflow.
 
-        Scores are persisted via the Submissions API, separate from the Workflow
-        Data. Score is associated with the same submission_uuid as this workflow
+        Allows for staff scores to bet set on a submission, with annotations to provide an audit trail if needed.
+        This method can be used for both required staff grading, and staff overrides.
 
         Args:
             score (dict): A dict containing 'points_earned' and
                 'points_possible'.
+            staff_name (string): The name of the staff member who is giving the score.
+            is_required (bool, default False): True if setting a required staff score.
+            is_override (bool, default False): True if staff is overriding a previous score.
+                -note that one and only one of is_required and is_override must be True
+            reason (string): An optional parameter specifying the reason for the staff grade. A default value
+                will be used in the event that this parameter is not provided.
 
         """
+        if is_required == is_override:
+            raise StaffAssessmentRequestError('Staff scores must provide one and only one of is_requred and is_override')
+        if not staff_name:
+            raise StaffAssessmentRequestError('Staff scores must provide the name of the staff member')
+        annotation_type = None
+        if is_required:
+            annotation_type = "Required staff score"
```

Comment:
I think it makes sense to make these constants and use underscores in the string constants to make the fact that they're a type a little clearer. Something like "required_staff_score".